traitlets/traitlets.py

```diff
@@ -64,7 +64,8 @@
 #-----------------------------------------------------------------------------
 # Basic classes
 #-----------------------------------------------------------------------------
-
+import warnings
+warnings.filterwarnings('error')
```

Comment:
This line, changing warnings to errors, should be removed for the final version of this PR.  This just makes it really easy to see where things break.